Sources/_StringProcessing/Engine/InstPayload.swift

```diff
@@ -211,11 +211,11 @@ extension Instruction.Payload {
     self.rawValue == 1
   }
 
-  init(bool: BoolRegister) {
-    self.init(bool)
+  init(bool: Bool) {
+    self.init(bool ? 1 : 0, 0)
   }
-  var bool: BoolRegister {
-    interpret()
+  var boolPayload: Bool {
+    interpret(as: TypedInt<Bool>.self) == 1
```

Comment:
Okay to switch this to a payload access? It doesn't look like we actually have a Boolean register.

Reply:
Sure, if there's no need to have a value that's stored and updated